packages/flutter/lib/src/services/text_input.dart

```diff
@@ -930,4 +909,77 @@ class TextInput {
     }());
     return true;
   }
+
+  MethodChannel _channel;
+
+  TextInputConnection _currentConnection;
+
+  Future<dynamic> _handleTextInputInvocation(MethodCall methodCall) async {
+    if (_currentConnection == null)
+      return;
+    final String method = methodCall.method;
+
+    // Reattach request needs to be handled regardless of the client ID.
+    if (method == 'TextInputClient.requestExistingInputState') {
+      assert(_currentConnection._client != null);
+      _attach(_currentConnection);
+      if (_lastTextEditingValue != null) {
```

Comment:
Does it seem odd that we cache the lastTextEditingValue, but don't cache the configuration?

Reply:
Just caching them all now. I think this is probably cleaner and it's no longer breaking.